embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/comms/delivery/v2/SchedulingService.kt

```diff
@@ -0,0 +1,111 @@
+package io.embrace.android.embracesdk.internal.comms.delivery.v2
+
+import io.embrace.android.embracesdk.internal.clock.Clock
+import io.embrace.android.embracesdk.internal.comms.api.ApiResponse
+import io.embrace.android.embracesdk.internal.worker.BackgroundWorker
+import io.embrace.android.embracesdk.internal.worker.ScheduledWorker
+import java.io.File
+import java.util.Queue
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.Future
+import java.util.concurrent.TimeUnit
+
+interface SchedulingService {
+    fun checkPendingPayloads()
+}
+
+internal class SchedulingServiceImpl(
+    private val cacheService: CacheService2,
+    private val executionService: ExecutionService,
+    private val schedulingWorker: ScheduledWorker,
+    private val worker: BackgroundWorker,
+    private val clock: Clock
+) : SchedulingService {
+
+    private val retryQueue: Queue<File> = ConcurrentLinkedQueue()
+
+    override fun checkPendingPayloads() {
+        schedulingWorker.submit {
+            var ready = readyToSend()
+            while (ready) {
+                runCatching {
+                    getNextReadyPayload()?.let { payload ->
+                        val response = send(payload).get()
```

Comment:
Not keen on this `.get()`. I think the approach of delegating to another worker is good & enables future multi-threaded request sending, but I don't think it makes sense to block this thread here

Reply:
Ah, I messed up the logic here. I meant to not block on the current thread, but instead do that in the `send()` and the the future that it returns. I'll fix that up